examples/todo/src/__tests__/acceptance/todo.acceptance.ts

```diff
@@ -189,7 +189,7 @@ describe('TodoApplication', () => {
 
     await client
       .get('/todos')
-      .query({filter: {where: {isComplete: false}}})
+      .query({filter: encodeURIComponent(JSON.stringify({where: {isComplete: false}})) })
```

Comment:
Ideally, both flavors should be supported:
 - `filter: {where: {isComplete: false}}` 
 - `filter: encodeURIComponent(JSON.stringify(...))`

Instead of changing this existing test, we should add a new test to verify the other flavor. I believe these two flavors are already supported now, in which case it would be great to have tests to verify all different ways how to provide the `filter` argument.

Reply:
(This code change & discussion thread is directly related to my previous comment about backwards compatibility.)

Reply:
An idea to consider & try: can we mix `deepObject` style with `json` style in a single parameter? Something along the following lines:

```ts
    object: function(
      name: string,
      schema: SchemaObject | ReferenceObject = {
        type: 'object',
        additionalProperties: true,
      },
      spec?: Partial<ParameterObject>,
    ) {
      schema = {
        type: 'object',
        ...schema,
      };
      return param({
        name,
        in: 'query',

        // deepObject
        style: 'deepObject',
        explode: true,

        // new addition - json encoding
        content: {
           "application/json": {
             schema
           }
         },

         // the rest
        schema,
         ...spec,
       });
     },
```

Reply:
+1 Let's make sure the default signature `filter: {where: {isComplete: false}}` still works.

Reply:
> Ideally, both flavors should be supported:
> 
> * `filter: {where: {isComplete: false}}`
> * `filter: encodeURIComponent(JSON.stringify(...))`
>

@bajtos while using super-test both the above are the same , ie, using `encodeURIComponent(JSON.stringify(` for super test is redundant, while for other http client invocations it is required. sending just `filter: {where: {isComplete: false}}` in url will not work.
I suppose you are asking to support exploded query params like `filter[limit]=1` 

> Instead of changing this existing test, we should add a new test to verify the other flavor. I believe these two flavors are already supported now, in which case it would be great to have tests to verify all different ways how to provide the `filter` argument.

yes both the exploded and complex objects are already supported in loopback, even without the changes in this PR. The suggested changes in this PR is only for API Explorer to support complex objects.

Reply:
@bajtos @jannyHou I have now implemented your suggestions to modify existing decorator and also have backward compatibility, ie, support both exploded query params and complex objects in query params.

Reply:
@bajtos @jannyHou , to be clear , currently loopback supports both exploded and uri encoded complex objects in query params, even without the changes suggested in this PR. The purpose of this PR is to ONLY fix API Explorer to support complex objects in query params.

Reply:
@bajtos @jannyHou , I have tested , for example apps API Explorer  is working fine now for complex objects in query params.